packages/loader/container-loader/src/container.ts

```diff
@@ -802,39 +798,54 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
         const previousContextState = await this.context.snapshotRuntimeState();
         this.context.dispose();
 
-        let snapshot: ISnapshotTree | undefined;
-        const blobs = new Map();
-        if (previousContextState.snapshot !== undefined) {
-            snapshot = await buildSnapshotTree(previousContextState.snapshot.entries, blobs);
-
-            /**
-             * Should be removed / updated after issue #2914 is fixed.
-             * There are currently two scenarios where this is called:
-             * 1. When a new code proposal is accepted - This should be set to true before `this.loadContext` is
-             * called which creates and loads the ContainerRuntime. This is because for "read" mode clients this
-             * flag is false which causes ContainerRuntime to create the internal components again.
-             * 2. When the first client connects in "write" mode - This happens when a client does not create the
-             * Container in detached mode. In this case, when the code proposal is accepted, we come here and we
-             * need to create the internal data stores in ContainerRuntime.
-             * Once we move to using detached container everywhere, this can move outside this block.
-             */
-            this._existing = true;
+        const codeDetails = this.getCodeDetailsFromQuorum();
+
+        // don't fire this event if we are transitioning from a null runtime to a real runtime
```

Comment:
Code in Bohemia wanted an event for transitioning from null to non-null runtime.  I'd guess that for as long as we have null runtime scenarios, we want to raise events for transitions.

Reply:
they still get the context changed as before, my guess is that they don't need it now that we have detached container, if that's not the case i'd like to understand it more

Reply:
i'd like to get rid of null runtime. it had to be exposed before detached, but soon it should be safe to assume all containers have a code proposal, from detached, or previous propose. So we can wait on loading any context until we see a valid code proposal, and get rid of null context